core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -587,6 +627,18 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
     }
   }
 
+  private[tools] object passThroughMirrorMakerMessageHandler extends MirrorMakerMessageHandler {
+    override def handle(record: BaseConsumerRecord): util.List[ProducerRecord[Array[Byte], Array[Byte]]] = {
+      val timestamp: java.lang.Long = if (record.timestamp == RecordBatch.NO_TIMESTAMP) null else record.timestamp
+      // It is assumed that we don't have message format V0 anymore at Linkedin
+      if (record.magic.equals(RecordBatch.MAGIC_VALUE_V1)) {
+        Collections.singletonList(new ProducerRecord(record.topic, null, timestamp, record.key, record.value, recordHeadersV1))
+      } else {
+        Collections.singletonList(new ProducerRecord(record.topic, null, timestamp, record.key, record.value, recordHeadersV2))
+      }
```

Comment:
you need a positive assertion in case you see something you dont expect. otherwise you'll get a sneaky bug